### PR TITLE
doc(user): Explain what Zebra does when it starts up

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

We want to explain to new users how Zebra starts up and connects to the network.

[Rendered](https://github.com/ZcashFoundation/zebra/blob/zebrad-startup-doc/book/src/user/startup.md)

## Solution

- Explain how Zebra starts up, based on the startup logs

## Review

Anyone can review this PR, we worked through the doc in a meeting.

### Reviewer Checklist

  - [ ] Rendering works

### CODE REVIEW

File: book/src/user/startup.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+zebra_state::service::finalized_state: loaded Zebra state cache tip=None
+zebra_state::service: created new read-only state service
+...
+```
+
+### Peer Handling
+
+#### Discover Initial Peers
+
+Zebra queries DNS seeders for initial peer IP addresses, then tries to connect to `network.peerset_initial_target_size` initial peers.
```
Comment (oxarbitrage) [inline book/src/user/startup.md:69]:
```suggestion
Zebra queries [DNS seeders](https://github.com/ZcashFoundation/dnsseeder) for initial peer IP addresses, then tries to connect to `network.peerset_initial_target_size` initial peers.
```

File: book/src/user/startup.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+Zebra also has an ongoing sync progress task, which logs progress towards the tip every minute.
+
+```
+zebrad::commands::start: spawned initial Zebra tasks
+zebrad::components::sync: starting genesis block download and verify
+zebrad::commands::start: initial sync is waiting to download the genesis block sync_percent=0.000 % current_height=None
+sync:checkpoint: zebra_consensus::checkpoint: verified checkpoint range block_count=1 current_range=(Unbounded, Included(Height(0)))
+sync:try_to_sync: zebrad::components::sync: starting sync, obtaining new tips state_tip=Some(Height(0))
+sync:try_to_sync: zebrad::components::sync: extending tips tips.len=1 in_flight=499 lookahead_limit=2000 state_tip=Some(Height(0))
+sync:try_to_sync:obtain_tips:checkpoint: zebra_consensus::checkpoint: verified checkpoint range block_count=400 current_range=(Excluded(Height(0)), Included(Height(400)))
```
Comment (oxarbitrage) [inline book/src/user/startup.md:184]: I think we can add `...` here and then show a line that prints more than `0.000%` to make the progress more clear.

**Review by oxarbitrage — DISMISSED** [review]
Looks good to me. I made a few minor optional comments.

File: book/src/user/startup.md (hunk 1 above)
Comment (teor2345) [reply book/src/user/startup.md:69]: There are 4 different DNS seeders, and only 2 of them run the Foundation's software at that link.

Do you think we could link to this definition instead? Or would it be confusing to use the Bitcoin definition?
https://btcinformation.org/en/glossary/dns-seed

File: book/src/user/startup.md (hunk 2 above)
Comment (teor2345) [reply book/src/user/startup.md:184]: Thanks, that's a good idea:

```suggestion
sync:try_to_sync:obtain_tips:checkpoint: zebra_consensus::checkpoint: verified checkpoint range block_count=400 current_range=(Excluded(Height(0)), Included(Height(400)))
...
zebrad::commands::start: estimated progress to chain tip sync_percent=0.537 % current_height=Height(9119) remaining_sync_blocks=1687657 time_since_last_state_block=PT0S
...
```

File: book/src/user/startup.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+```
+zebrad::commands::start: initializing network
+open_listener{addr=127.0.0.1:8233}: zebra_network::peer_set::initialize: Trying to open Zcash protocol endpoint at 127.0.0.1:8233...
+open_listener{addr=127.0.0.1:8233}: zebra_network::peer_set::initialize: Opened Zcash protocol endpoint at 127.0.0.1:8233
+zebra_network::address_book_updater: starting the address book updater
+add_initial_peers: zebra_network::config: resolved seed peer IP addresses seed="mainnet.seeder.zfnd.org:8233" remote_ip_count=25
+...
+add_initial_peers: zebra_network::peer_set::initialize: limiting the initial peers list from 112 to 25
+```
+
```
Comment (oxarbitrage) [inline book/src/user/startup.md:81]:
```suggestion
**DNS Seeder**: A DNS server which returns IP addresses of full nodes on the Zcash network to assist in peer discovery. [Zcash Foundation dnsseeder](https://github.com/ZcashFoundation/dnsseeder)  

```

File: book/src/user/startup.md (hunk 1 above)
Comment (oxarbitrage) [reply book/src/user/startup.md:69]: Maybe something more as https://github.com/ZcashFoundation/zebra/pull/4502/files#r885601065 instead.

**Review by oxarbitrage — APPROVED** [review]
Thanks!, looks great.

**Pull request merged**